src/ORM/SoftDeletable/SoftDeletableSubscriber.php

```diff
@@ -48,9 +35,10 @@ public function onFlush(OnFlushEventArgs $args)
         $em  = $args->getEntityManager();
         $uow = $em->getUnitOfWork();
 
+        /** @var SoftDeletableInterface $entity */
         foreach ($uow->getScheduledEntityDeletions() as $entity) {
             $classMetadata = $em->getClassMetadata(get_class($entity));
-            if ($this->isSoftDeletable($classMetadata)) {
+            if (is_subclass_of($classMetadata->getName(), 'Knp\DoctrineBehaviors\Model\SoftDeletable\SoftDeletableInterface')) {
```

Comment:
Wouldn't you prefer basing yourself on the *Properties trait to trigger the mapping ? Using the interface might cause problems in case of inheritance (CF #317).

What do you think about the pattern I used in my PRs (e.g. https://github.com/KnpLabs/DoctrineBehaviors/pull/322/files#diff-9af172be101c124fe7673df513efc25aL296)